src/arrayMethodReduce.js

```diff
@@ -5,7 +5,15 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const isInitialValue = (arguments.length === 2);
+
+    let accumulator = isInitialValue ? initialValue : this[0];
+
+    for (let i = isInitialValue ? 0 : 1; i < this.length; i++) {
+      accumulator = callback(accumulator, this[i], i, this);
+    }
```

Comment:
the code will be easier to read if you create a variable for start value
```suggestion

		const start = isInitialValue ? 0 : 1;
		
    for (let i = start; i < this.length; i++) {
      accumulator = callback(accumulator, this[i], i, this);
    }
```

Reply:
I agree with Nazar, let's make it clear